round_robin.go

```diff
@@ -33,5 +33,5 @@ func New(urls ...*url.URL) (RoundRobin, error) {
 // Next returns next address
 func (r *roundrobin) Next() *url.URL {
 	n := atomic.AddUint32(&r.next, 1)
-	return r.urls[(int(n)-1)%len(r.urls)]
+	return r.urls[(int(n)%len(r.urls)]
```

Comment:
By the way, this implementation sets the initial value of the index to 1 and the test fails, so how about the following implementation?

```suggestion
	n := atomic.AddUint32(&r.next, 1) - 1
	if n < 0 {
		n = 0
	}
	return r.urls[int(n)%len(r.urls)]
```

Reply:
It's ok when you considering the test case. I use `n := atomic.AddUint32(&r.next, 1) return r.urls[int(n)%len(r.urls)]` in my project.

Appreciated your project.